app/models/submission/validations_by_template_name.rb

```diff
@@ -65,5 +80,94 @@ def validate_consistent_column_value(column_header)
       )
     end
   end
-  # rubocop:enable Metrics/MethodLength,Metrics/AbcSize
+  # rubocop:enable Metrics/MethodLength
+
+  def validate_samples_per_pool_for_labware
```

Comment:
add method comments for new methods. 
codepilot is pretty good at doing most of it if you highlight the method and ask it to 'add a method comment'